### ISSUE TITLE
Only run middleware on "ember" requests

### ISSUE THREAD

**Issue by seanpdoyle** [rwz/ember-cli-rails#139]
Could we provide a hook (either via the routes or a block in the middleware declaration) to determine if a request will be handled by the middleware?

I think the middleware is running on *all* requests. 

In testing, precompiling ember for `/api/` requests dramatically slows down an otherwise rapid feedback cycle.

What do you think? (<small>warning: pseudocode approaching</small>)

```ruby
config.use EmberCLI::ReloadMiddleware do |request|
  request.url =~ %r{/frontend/foo}
end
```

**Comment by rwz** [comment]
We've tried that in the past and failed. The problem was the assets pipeline (as always).

When you're trying to request a page that has `javascript_include_tag` or `stylesheet_link_tag` in it and the files aren't actually available by the time page is rendered, rails falls back to using old-style assets paths, such as `/javascripts/frontend/frontend.js` and `/stylesheets/frontend/frontend.css`.

Because of this behavior currently we need precompile ember assets before first request hits the app.

**Comment by rwz** [comment]
if you have any ideas about how to make it work, feel free to reopen this issue :)

**Comment by seanpdoyle** [comment]
@rwz I'm thinking about `/api` routes that don't render pages at all.

Wouldn't a controller than calls `render json: ` work without precompiling assets?

**Comment by rwz** [comment]
It would. But there's no way to tell for sure which urls require assets and which don't

**Comment by seanpdoyle** [comment]
Would it make sense to allow for whitelisting URLs to skip?

**Comment by seanpdoyle** [comment]
i.e. I might not know all the routes that my ember app could be served from, but it's likely that i know that `/api` or subdomain `api` won't serve asset tags

**Comment by rwz** [comment]
@seanpdoyle Ok, I've created a new branch called `runner`. It supports `enable` proc option to customize which paths of your app needs certain ember app to be compiled. It also should address the issue of dying background process. Give it a spin and report if it works for you.

**Comment by seanpdoyle** [comment]
![screen shot 2015-04-06 at 10 12 23 am](https://cloud.githubusercontent.com/assets/2575027/7005382/8563b270-dc45-11e4-9c85-05724bfb3f12.png)

:disappointed: could you drop the `.freeze` on the `VERSION`?

**Comment by slashmili** [comment]
How does this enable options work?

I have this config for EmberCLI

    EmberCLI.configure do |c|
      c.app :myapp,
        path: "ui/myapp",
        enable: -> path { !path.starts_with?("/api/") }
    end

And I have these routes:

    Rails.application.routes.draw do
      root 'application#index'
      scope "/api/v1" do
        resources :users
      end
    end

When I open http://localhost:3000/api/v1/users it goes to Ember and it complains that 

    Uncaught UnrecognizedURLError: /api/v1/users

**Comment by seanpdoyle** [comment]
closed by #145
